File: 2º/PDOO/ProyectoRUBY/lib/controlador.rb
# encoding:utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
#require_relative"operacion_inmobiliaria.rb"
module Civitas
  class Controlador
    def initialize(juego, vista)
      @juego=juego
      @vista=vista
    end
    def juega
      @vista.setCivitasJuego(@juego)
      while(@juego.finalDelJuego==false)
        @vista.actualizarVista        
        @vista.pausa
        operacion=@juego.siguientePaso
        @vista.mostrarSiguienteOperacion(operacion)
        if(operacion==Operaciones_juego::PASAR_TURNO)then
          puts("*******************")
          @vista.mostrarEventos
          puts("*******************")
        end
        if(@juego.finalDelJuego==false)then
          if(operacion==Operaciones_juego::COMPRAR)then
           # if(@juego.getCasillaActual.tipo==TipoCasilla::CALLE)then
              resul=@vista.comprar
              if(resul==Respuestas::SI)then
                @juego.comprar
              end
            #else
            #  puts("Esta casilla no se puede comprar")
            #end  
            @juego.siguientePasoCompletado(operacion)####
          elsif(operacion==Operaciones_juego::GESTIONAR)
            @vista.gestionar
            gesti=@vista.getGestion
            propi=@vista.getPropiedad
            operacionIm=OperacionInmobiliaria.new(gesti, propi)
            if(operacionIm.getGestion()==GestionesInmobiliarias::CANCELAR_HIPOTECA)then
              @juego.cancelarHipoteca(propi)
            elsif(operacionIm.getGestion()==GestionesInmobiliarias::CONSTRUIR_CASA)then
              @juego.construirCasa(propi)
            elsif(operacionIm.getGestion()==GestionesInmobiliarias::CONSTRUIR_HOTEL)then  
              @juego.construirHotel(propi)
            elsif(operacionIm.getGestion()==GestionesInmobiliarias::HIPOTECAR)then
              @juego.hipotecar(propi)
            elsif(operacionIm.getGestion()==GestionesInmobiliarias::VENDER)then
              @juego.vender(propi)
            elsif(operacionIm.getGestion()==GestionesInmobiliarias::TERMINAR)then
              @juego.siguientePasoCompletado(operacion)
            end  
          elsif(operacion==Operaciones_juego::SALIR_CARCEL)  
            tiposal=@vista.salirCarcel
            if(tiposal==SalidasCarcel::PAGANDO)then
              @juego.salirCarcelPagando
            elsif(tiposal==SalidasCarcel::TIRANDO)then 
              @juego.salirCarcelTirando
            end
          end
        end
      end
      rerank=@juego.ranking
      if(rerank.size==4)then
        puts("El resultado del juego es el siguiente: 1º#{rerank[0].nombre}---2º#{rerank[1].nombre}---3º#{rerank[2].nombre}---4º#{rerank[3].nombre}")
      elsif(rerank.size==3)then
        puts("El resultado del juego es el siguiente: 1º#{rerank[0].nombre}---2º#{rerank[1].nombre}---3º#{rerank[2].nombre}")
      elsif(rerank.size==2)then
        puts("El resultado del juego es el siguiente: 1º#{rerank[0].nombre}---2º#{rerank[1].nombre}")
      else
        puts("El ranking del final del juego en controlador ha fallado")
      end
    end
  end
end

File: 2º/PDOO/ProyectoJAVA/src/GUI/Controlador.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package GUI;

import civitas.CivitasJuego;
import civitas.GestionesInmobiliarias;
import civitas.OperacionesJuego;
import civitas.OperacionInmobiliaria;
import civitas.SalidasCarcel;
import civitas.GestionesInmobiliarias;

/**
 *
 * @author nikitastetskiy
 */
public class Controlador {
    private CivitasJuego juego;
    private CivitasView vista;
    
    public Controlador(CivitasJuego juego, CivitasView vista){
        this.juego = juego;
        this.vista = vista;
    }
    
    public void juega(){
        vista.setCivitasJuego(juego);
        while(!juego.finalDelJuego(0)){
            vista.actualizarVista();
           // vista.pausa();
            OperacionesJuego operacion = juego.siguientePaso();
            vista.mostrarSiguienteOperacion(operacion);
            if(operacion != OperacionesJuego.PASAR_TURNO)
                vista.mostrarEventos();
            if(!juego.finalDelJuego(0)){
                if(operacion == OperacionesJuego.COMPRAR){
                    if(vista.comprar() == Respuestas.SI){
                        juego.comprar();
                    }
                    juego.siguientePasoCompletado(operacion);
                }else if(operacion == OperacionesJuego.GESTIONAR){
                    vista.gestionar();
                    /*OperacionInmobiliaria op_inmobiliaria = new OperacionInmobiliaria(GestionesInmobiliarias.values()[vista.getGestion()], vista.getPropiedad());
                    juego.siguientePasoCompletado(operacion);*/
                    GestionesInmobiliarias gest = GestionesInmobiliarias.values()[vista.getGestion()];
                    int ip = vista.getPropiedad();
                    OperacionInmobiliaria op = new OperacionInmobiliaria(gest, ip);
                    if(op.getGestion()==GestionesInmobiliarias.VENDER){
                        juego.vender(vista.getPropiedad());
                    }else if(op.getGestion()==GestionesInmobiliarias.HIPOTECAR){
                        juego.hipotecar(vista.getPropiedad());
                    }else if(op.getGestion()==GestionesInmobiliarias.CANCELAR_HIPOTECA){
                        juego.cancelarHipoteca(vista.getPropiedad());
                    }else if(op.getGestion()==GestionesInmobiliarias.CONSTRUIR_CASA){
                        juego.construirCasa(vista.getPropiedad());
                    }else if(op.getGestion()==GestionesInmobiliarias.CONSTRUIR_HOTEL){
                        juego.construirHotel(vista.getPropiedad());
                    }else if(op.getGestion()==GestionesInmobiliarias.TERMINAR){
                        juego.siguientePasoCompletado(operacion);
                    }
                }else if(operacion == OperacionesJuego.SALIR_CARCEL){
                    SalidasCarcel salirCarcel = vista.salirCarcel();
                    if(salirCarcel == SalidasCarcel.PAGANDO)
                        juego.salirCarcelPagando();
                    else if(salirCarcel == SalidasCarcel.TIRANDO)
                        juego.salirCarcelTirando();
                    juego.siguientePasoCompletado(operacion);
                }
            }
        }
        System.out.println(juego.ranking());
    }
}
